### DISCUSSION TITLE
Updated from RHF 7.5.2 to 7.6.6 and now form isn't valid

### DISCUSSION THREAD

**Discussion by NickR49** [react-hook-form/react-hook-form#5343] [Q&A] [answered]
I have a form which previously only needed the two required fields to be entered for it to become valid (though there are other fields).
When I updated from RHF 7.5.2 to 7.6.6 entering those two fields no longer sets the form to a valid state.
I can see there are no errors against any of the fields. 
Is there some way to tell what is causing the form to be considered invalid?

**Comment by bluebill1049** [comment, marked as answer]
It could be a bug, any chance you could share a codesandbox? happy to investigate the root cause.

**Reply by NickR49** [reply]
I just outputted `formState.errors` and it shows no errors. I use `reValidateMode: 'onChange'`.
I'll see if I can get this into CSB for you.

**Reply by bluebill1049** [reply]
hmm.... could be fields which considered to having validation but not validated. we track isValid not with errors, but inputs have validation against and are they get validated.

**Reply by NickR49** [reply]
Ah okay, there are other fields that have validation but they only need validation if a certain selection is made. I think maybe the form previously considered those fields valid until validation was invoked and failed - innocent until proven guilty!

**Reply by NickR49** [reply]
I might have to rethink how I validate conditional fields

**Reply by NickR49** [reply]
In my particular example the conditional field which appears to be stopping the form from being valid is a select field which is required but has a value already assigned on initialisation. It seems that that field needs to be touched in order for the form to be flagged as valid. As soon as I change that field the form becomes valid (the field is now dirty). If I change the select field back to what it was (no longer dirty) the form remains valid. Seems odd . . .

I'll figure out a CSB that closely reproduces this but it'll be in the morning (NZT :))

**Reply by NickR49** [reply]
Okay this is super weird. I have a radio group. Radio 1 has a text field. Radio 2 has a text field. If I select radio 1 or 2 then both fields retain their form values.

But if I add a disabled attribute to the first text input and disable it when radio 1 is not selected - then the first text field only retains its form value when it is NOT selected ie. when it is disabled. I'll create a CSB for this.

**Reply by bluebill1049** [reply]
Is this related to this https://github.com/react-hook-form/react-hook-form/pull/5350#issuecomment-849245460?

**Reply by NickR49** [reply]
I think it could be related. I created a CSB that reproduced the issue but somewhere along the way as I simplified the example it started working correctly so it's not much help I'm afraid. Here it is at any rate - https://codesandbox.io/s/rhf7-disabled-field-issue-qt32z?file=/src/App.tsx

**Reply by bluebill1049** [reply]
Lets try to get a beta version on the PR above and see if that resolves your issue. PR  build is still failing and should get fixed tmr 😌

**Reply by bluebill1049** [reply]
can you try out this beta: `7.6.8-beta.0`?

**Reply by NickR49** [reply]
I've tried 7.6.8 and the issue with conditional fields (that haven't been validated - probably because they're disabled) stopping the form from being valid is still there. I've gone back and found that it was still working in 7.5.3 but stopped working in 7.6.0.
The only fix I've found so far is to turn off the validation for conditional fields but this isn't ideal.

I think the documentation for `formState.isValid` might be a bit misleading as it's not entirely true - "Set to true if the form doesn't have any errors"

**Reply by bluebill1049** [reply]
someone just reported issue with a csb: https://github.com/react-hook-form/react-hook-form/issues/5366 I think it may related.

**Reply by bluebill1049** [reply]
I am addressing this isValid issue: https://github.com/react-hook-form/react-hook-form/pull/5415 I am removing the internal state to track isValid.

**Reply by bluebill1049** [reply]
let me know if this beta is fixing your issue: v7.7.2-beta.0

**Reply by NickR49** [reply]
Hi Bill, I've just tested with version 7.7.1 and `shouldUnregister: true` fixes my problem :)
I think that because the fields in the non-selected options are disabled that they are being unregistered?
So then the validation on those non-selected options are no longer stopping the form from being valid.
(The problem was the `required: true` on the disabled select component was stopping the form from being valid)

https://user-images.githubusercontent.com/43474104/120396765-3b700680-c38b-11eb-84b5-63ce3952f7f4.mp4

I've also just tested with v7.7.2-beta.0 and `shouldUnregister: false` and this also works :)
Thank you so much for your help!

**Reply by bluebill1049** [reply]
awesome! I will release the improvement this weekend.
